backup_function.tf

```diff
@@ -1,127 +1,138 @@
-resource "aws_iam_role" "ebs_backup_role" {
-  name = "ebs_backup_role"
-
-  assume_role_policy = <<EOF
-{
-  "Version": "2012-10-17",
-  "Statement": [
-    {
-      "Action": "sts:AssumeRole",
-      "Principal": {
-        "Service": "lambda.amazonaws.com"
-      },
-      "Effect": "Allow",
-      "Sid": ""
+data "aws_iam_policy_document" "default" {
+  statement {
+    sid = ""
+
+    principals {
+      type = "Service"
+
+      identifiers = [
+        "lambda.amazonaws.com",
+      ]
     }
-  ]
-}
-EOF
+
+    actions = [
+      "sts:AssumeRole",
+    ]
+  }
 }
 
-resource "aws_iam_role_policy" "ebs_backup_policy" {
-  name = "ebs_backup_policy"
-  role = "${aws_iam_role.ebs_backup_role.id}"
-
-  policy = <<EOF
-{
-    "Version": "2012-10-17",
-    "Statement": [
-        {
-            "Effect": "Allow",
-            "Action": ["logs:*"],
-            "Resource": "arn:aws:logs:*:*:*"
-        },
-        {
-            "Effect": "Allow",
-            "Action": "ec2:Describe*",
-            "Resource": "*"
-        },
-        {
-            "Effect": "Allow",
-            "Action": [
-                "ec2:CreateSnapshot",
-                "ec2:DeleteSnapshot",
-                "ec2:CreateTags",
-                "ec2:ModifySnapshotAttribute",
-                "ec2:ResetSnapshotAttribute"
-            ],
-            "Resource": ["*"]
-        }
+data "aws_iam_policy_document" "ami_backup" {
+  statement {
+    actions = [
+      "logs:*",
+    ]
+
+    resources = [
+      "arn:aws:logs:*:*:*",
+    ]
+  }
+
+  statement {
+    actions = [
+      "ec2:*",
```

Comment:
Why are we giving carte blanche access? Please use explicit permissions and not `*`